src/tools/simeng/main.cc

```diff
@@ -197,107 +162,46 @@ int main(int argc, char** argv) {
     //                                  17, 4, 3, 22, 117, 11, 4,  12, 10, 18};
     // memcpy(memory, memoryValues.data(), memoryValues.size() * sizeof(int));
 
-    process = std::make_unique<simeng::kernel::LinuxProcess>(
-        simeng::span<char>(reinterpret_cast<char*>(hex), sizeof(hex)), config);
+    coreInstance->createProcess(
+        simeng::span<char>(reinterpret_cast<char*>(hex), sizeof(hex)));
   }
 
-  // Read the process image and copy to memory
-  auto processImage = process->getProcessImage();
-  size_t processMemorySize = processImage.size();
-  char* processMemory = new char[processMemorySize]();
-  std::copy(processImage.begin(), processImage.end(), processMemory);
-
-  uint64_t entryPoint = process->getEntryPoint();
-
-  // Create the OS kernel with the process
-  simeng::kernel::Linux kernel;
-  kernel.createProcess(*process.get());
-
-  simeng::FlatMemoryInterface instructionMemory(processMemory,
-                                                processMemorySize);
-
-  // Create the architecture, with knowledge of the kernel
-  std::unique_ptr<simeng::arch::Architecture> arch =
-      std::make_unique<simeng::arch::aarch64::Architecture>(kernel, config);
-
-  auto predictor = simeng::GenericPredictor(config);
-  auto config_ports = config["Ports"];
-  std::vector<std::vector<uint16_t>> portArrangement(config_ports.size());
-  // Extract number of ports
-  for (size_t i = 0; i < config_ports.size(); i++) {
-    auto config_groups = config_ports[i]["Instruction-Group-Support"];
-    // Extract number of groups in port
-    for (size_t j = 0; j < config_groups.size(); j++) {
-      portArrangement[i].push_back(config_groups[j].as<uint16_t>());
-    }
+  // Get simualtion mode and data memory interface type
+  simeng::SimulationMode mode = coreInstance->getSimulationMode();
+  simeng::MemInterfaceType L1Dtype = simeng::MemInterfaceType::Flat;
```

Comment:
Why can't we have MemInterfaceType member variable inside `CoreInstance` which gets populated in the Constructor, given that we already get `SimulationMode` using the `getSimulationMode` getter? And expose it (if necessary) via a getter?

Reply:
This would reduce the flexibility of the possible configurations. Whilst we currently always use a set memory interface type for each simulation mode, there's nothing to say we wouldn't use differing/new interface types in the future.